### ISSUE TITLE
Sonarr cannot write to CIFS mount (but Radarr can)

### ISSUE THREAD

**Issue by shuchuka** [Sonarr/Sonarr#6425]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Current Behavior

Adding a root folder to Sonarr that is located in a CIFS mounted directory with the proper permissions raises an error `Unable to add root folder - Folder '/mnt/library/tv/' is not writable by user 'sonarr'`. More details about the setup below, but the reason I think this is a bug is that Sonarr and Radarr are set up in an identical way, yet Radarr is able to add the same directory as a root folder. Running read/write/creation/deletion commands as the user `sonarr` also succeeds without issue.

### Expected Behavior

The directory should be added as a root folder in Sonarr.

### Steps To Reproduce

1. A fresh install of Raspbian Pi OS lite (32-bit)
2. Install Sonarr and Radarr using the manual instructions for Debian here:  https://wiki.servarr.com/radarr/installation/linux
    `uid=1002(sonarr) gid=1002(sonarr) groups=1002(sonarr),1003(media)`
    `uid=1001(radarr) gid=1001(radarr) groups=1001(radarr),1003(media)`
3. Mount local network share in `/etc/fstab` via
    `//<local_ip>/<remote_dir> /mnt/library cifs credentials=<cred_file>,uid=0,gid=1003,file_mode=0775,dir_mode=0775 0 0`
4. Add root folder `/mnt/library/tv` via UI in the browser

### Environment

```markdown
- OS: Raspbian 11.8
- Sonarr: 4.0.1.929
- Docker Install: No
- Using Reverse Proxy: No
- Browser: Firefox 121.0.1
- Database: Sqlite 3.34.1
```

### What branch are you running?

Main

### Trace Logs?

2024-01-25 20:32:50.9|Trace|Http|Req: 307 [GET] /api/v3/filesystem?path=&allowFoldersWithoutTrailingSlashes=true&includeFiles=false (from 192.168.1.102 Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:121.0) Gecko/20100101 Firefox/121.0)
2024-01-25 20:32:50.9|Trace|Http|Res: 307 [GET] /api/v3/filesystem?path=&allowFoldersWithoutTrailingSlashes=true&includeFiles=false: 200.OK (7 ms)
2024-01-25 20:32:50.9|Debug|Api|[GET] /api/v3/filesystem?path=&allowFoldersWithoutTrailingSlashes=true&includeFiles=false: 200.OK (7 ms)
2024-01-25 20:32:53.3|Trace|Http|Req: 308 [GET] /api/v3/filesystem?path=%2Fmnt%2F&allowFoldersWithoutTrailingSlashes=true&includeFiles=false (from 192.168.1.102 Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:121.0) Gecko/20100101 Firefox/121.0)
2024-01-25 20:32:53.4|Trace|Http|Res: 308 [GET] /api/v3/filesystem?path=%2Fmnt%2F&allowFoldersWithoutTrailingSlashes=true&includeFiles=false: 200.OK (83 ms)
2024-01-25 20:32:53.4|Debug|Api|[GET] /api/v3/filesystem?path=%2Fmnt%2F&allowFoldersWithoutTrailingSlashes=true&includeFiles=false: 200.OK (83 ms)
2024-01-25 20:32:55.4|Trace|Http|Req: 309 [GET] /api/v3/filesystem?path=%2Fmnt%2Flibrary%2F&allowFoldersWithoutTrailingSlashes=true&includeFiles=false (from 192.168.1.102 Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:121.0) Gecko/20100101 Firefox/121.0)2024-01-25 20:32:55.5|Trace|Http|Res: 309 [GET] /api/v3/filesystem?path=%2Fmnt%2Flibrary%2F&allowFoldersWithoutTrailingSlashes=true&includeFiles=false: 200.OK (125 ms)
2024-01-25 20:32:55.5|Debug|Api|[GET] /api/v3/filesystem?path=%2Fmnt%2Flibrary%2F&allowFoldersWithoutTrailingSlashes=true&includeFiles=false: 200.OK (125 ms)
2024-01-25 20:32:56.6|Trace|Http|Req: 310 [GET] /api/v3/filesystem?path=%2Fmnt%2Flibrary%2Ftv%2F&allowFoldersWithoutTrailingSlashes=true&includeFiles=false (from 192.168.1.102 Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:121.0) Gecko/20100101 Firefox/121.0)
2024-01-25 20:32:56.7|Trace|Http|Res: 310 [GET] /api/v3/filesystem?path=%2Fmnt%2Flibrary%2Ftv%2F&allowFoldersWithoutTrailingSlashes=true&includeFiles=false: 200.OK (123 ms)
2024-01-25 20:32:56.7|Debug|Api|[GET] /api/v3/filesystem?path=%2Fmnt%2Flibrary%2Ftv%2F&allowFoldersWithoutTrailingSlashes=true&includeFiles=false: 200.OK (123 ms)
2024-01-25 20:32:58.3|Trace|Http|Req: 311 [POST] /api/v3/rootFolder (from 192.168.1.102 Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:121.0) Gecko/20100101 Firefox/121.0)
2024-01-25 20:32:58.3|Trace|ConfigService|Using default config value for 'recyclebin' defaultValue:''
2024-01-25 20:32:58.5|Trace|DiskProviderBase|Directory '/mnt/library/tv/' isn't writable. Access to the path '/mnt/library/tv/sonarr_write_test.txt' is denied.
2024-01-25 20:32:58.5|Trace|SonarrErrorPipeline|Handling Exception
2024-01-25 20:32:58.5|Warn|SonarrErrorPipeline|Invalid request Validation failed:
 -- Path: Folder '/mnt/library/tv/' is not writable by user 'sonarr'
2024-01-25 20:32:58.5|Trace|Http|Res: 311 [POST] /api/v3/rootFolder: 400.BadRequest (195 ms)
2024-01-25 20:32:58.5|Debug|Api|[POST] /api/v3/rootFolder: 400.BadRequest (195 ms)

### Anything else?

Note that attempting to add the root folder leaves behind a file `sonarr_write_test.txt` of size 0 in the directory in question.

A few more details about what I've tried to diagnose the issue:
- I had a look at the origin of the error in the source code to see what both projects are doing behind the scenes, but the path validation code is virtually identical for both projects:
  https://github.com/Sonarr/Sonarr/blob/b64c52a846529233d796014e1537b96e8f48ff7f/src/NzbDrone.Common/Disk/DiskProviderBase.cs#L126-L143 vs https://github.com/Radarr/Radarr/blob/develop/src/NzbDrone.Common/Disk/DiskProviderBase.cs#L126-L143
- I logged file system access to the mounted directory with `inotifywait` to try and identify what happens behind the scene during the procedure (from hitting `Ok` in the `Add root folder` dialogue).
  With Sonarr, the following events are logged:
  ```
  /mnt/library/tv/ OPEN sonarr_write_test.txt
  /mnt/library/tv/ MODIFY sonarr_write_test.txt
  /mnt/library/tv/ CLOSE_WRITE,CLOSE sonarr_write_test.txt
  ```
  Radarr on the other hand produces:
  ```
  /mnt/library/tv/ OPEN radarr_write_test.txt
  /mnt/library/tv/ MODIFY radarr_write_test.txt
  /mnt/library/tv/ MODIFY radarr_write_test.txt
  /mnt/library/tv/ CLOSE_WRITE,CLOSE radarr_write_test.txt
  /mnt/library/tv/ DELETE radarr_write_test.txt
  ```
  So we can see that the same write test succeeds in Radarr but not in Sonarr.

**Comment by markus101** [comment]
Please use one of the support channels: [forums](https://forums.sonarr.tv/), [subreddit](https://www.reddit.com/r/sonarr/), [discord ](https://discord.sonarr.tv/), or [IRC](https://web.libera.chat/?channels=#sonarr) for support/questions.

**Comment by fryfrog** [comment]
Probably need `nobrl` in your smb mount options.

**Comment by hansenc0705** [comment]
@shuchuka What build of radarr are you using? I'm having a similar issue but both radarr and Sonarr are impacted.

**Comment by shuchuka** [comment]
> Probably need `nobrl` in your smb mount options.

Hi @fryfrog, really appreciate the response! I tried your suggestion and it did work, so thank you very much for that, but it does feel like a workaround. 

Props to @stevietv for looking into why it was working with Radarr but not with Sonarr and porting the fix 🎉 Can confirm the latest version `4.0.1.1014` is working out of the box now.

> @shuchuka What build of radarr are you using? I'm having a similar issue but both radarr and Sonarr are impacted.

I was using `5.2.6.8376` on the master branch.

**Comment by fryfrog** [comment]
Nice! Did you re-test w/o `nobrl`?
